commit e97c0ddf29e15f2d693dbb9f5dee148b5803e6c7
Author: Radha Mohan Chintakuntla <rchintakuntla@cavium.com>
Date:   Wed Aug 20 15:10:58 2014 -0700

    UBUNTU: SAUCE: (no-up) arm64: gicv3: its: Increase FORCE_MAX_ZONEORDER for Cavium ThunderX
    
    BugLink: http://bugs.launchpad.net/bugs/1558828
    
    In case of ARCH_THUNDER, there is a need to allocate the GICv3 ITS table
    which is bigger than the allowed max order. So we are forcing it only in
    case of 4KB page size.
    
    Signed-off-by: Radha Mohan Chintakuntla <rchintakuntla@cavium.com>
    Signed-off-by: Robert Richter <rrichter@cavium.com>
    [ dannf: Depend on ARM64_4K_PAGES instead of !ARM64_64K_PAGES now that
      16K pages are available ]
    Signed-off-by: dann frazier <dann.frazier@canonical.com>
    Signed-off-by: Tim Gardner <tim.gardner@canonical.com>

diff --git a/arch/arm64/Kconfig b/arch/arm64/Kconfig
index 03383f3..91fe53a 100644
--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@ -698,6 +698,7 @@ config XEN
 config FORCE_MAX_ZONEORDER
 	int
 	default "14" if (ARM64_64K_PAGES && TRANSPARENT_HUGEPAGE)
+	default "13" if (ARCH_THUNDER && ARM64_4K_PAGES)
 	default "12" if (ARM64_16K_PAGES && TRANSPARENT_HUGEPAGE)
 	default "11"
 	help
